src/shared/sources/nz/index.js

```diff
@@ -0,0 +1,80 @@
+const assert = require('assert')
+const maintainers = require('../_lib/maintainers.js')
+const parse = require('../_lib/parse.js')
+const transform = require('../_lib/transform.js')
+
+const UNASSIGNED = '(unassigned)'
+
+const country = 'iso1:NZ'
+
+const schemaKeysByHeadingFragment = {
+  'total': null,
+  'in the last 24 hours': null,
+  'deceased': 'deaths',
+  'recovered': 'recovered',
+  'active': 'cases',
+  'dhb': 'state'
+}
+
+module.exports = {
+  aggregate: 'state',
+  country,
+  friendly: {
+    name: 'New Zealand Government Ministry of Health',
+    url: 'https://www.health.govt.nz'
+  },
+  maintainers: [ maintainers.camjc ],
+  scrapers: [
+    {
+      startDate: '2020-04-07',
+      crawl: [
+        {
+          type: 'page',
+          data: 'table',
+          url:
+            'https://www.health.govt.nz/our-work/diseases-and-conditions/covid-19-novel-coronavirus/covid-19-current-situation/covid-19-current-cases'
+        }
+      ],
+      scrape ($, date, { assertTotalsAreReasonable, buildGetIso2FromName, getSchemaKeyFromHeading, normalizeTable }) {
+        const getIso2FromName = buildGetIso2FromName({ country })
+        const normalizedTable = normalizeTable({ $, tableSelector: 'table:contains("Total cases by DHB")' })
+
+        const headingRowIndex = 0
+        const dataKeysByColumnIndex = []
+        normalizedTable[headingRowIndex].forEach((heading, index) => {
+          dataKeysByColumnIndex[index] = getSchemaKeyFromHeading({ heading, schemaKeysByHeadingFragment })
+        })
+
+        const dataRows = normalizedTable.slice(1, -1)
+
+        const statesCount = 20
+        assert.equal(dataRows.length, statesCount, 'Wrong number of rows found')
+
+        const states = []
+        dataRows.forEach((row) => {
+          const stateData = {}
+          row.forEach((value, columnIndex) => {
+            const key = dataKeysByColumnIndex[columnIndex]
+            stateData[key] = value
+          })
+
+          states.push({
+            state: getIso2FromName(stateData.state.replace('Capital and Coast', UNASSIGNED).replace('Counties Manukau', UNASSIGNED).replace('Hutt Valley', UNASSIGNED).replace('Lakes', UNASSIGNED).replace('Mid Central', UNASSIGNED).replace('Nelson Marlborough', UNASSIGNED).replace('South Canterbury', UNASSIGNED).replace('Southern', UNASSIGNED).replace('Tairawhiti', UNASSIGNED).replace('Wairarapa', UNASSIGNED).replace('Waitemata', UNASSIGNED)),
```

Comment:
All the un-matching states. How best to handle this? Should we merge as-is for now and see how to align later? (At least this would give us the counts for NZ country)

Reply:
Yes, we should merge as-is, it's better to have something than nothing, and we can iterate.